### ISSUE TITLE
Essentials - MediaPicker add method to select multiple images

### ISSUE THREAD

**Issue by janseris** [dotnet/maui#6903]
### Description

Currently for images/photos, there is `MediaPicker` class with `PickPhotoAsync` and `CapturePhotoAsync` methods available.
`PickPhotoAsync ` only selects 1 image at a time from gallery.
However, for `FilePicker`, there is `PickMultipleAsync `method available.

It should be possible to select multiple images, too.

Temporary workaround:
- probably via `FilePicker.PickMultipleAsync` with some image types restriction but that is _unneccessary complexity_

### Public API Changes

`MediaPicker` class: add `PickMultiplePhotosAsync` or similar method

### Intended Use-Case

pick multiple images from gallery

**Comment by janseris** [comment]
@AbstractionsAs I suggest you add like reaction to the post to push it upwards in priority. A lot of .NET GitHub works like this.

**Comment by johanlangevillecab** [comment]
It would also be nice to have a CapturePhotosAsync to take multiple photos without closing the camera

**Comment by ctigrisht** [comment]
I fully support this, it should have been a basic feature

**Comment by nathanjeynes** [comment]
This is needed and was available in the Xamarin forms essentials, not sure why this has been left out but its a key feature that many apps rely on.

**Comment by antiGithub** [comment]
I would also need this feature very badly

**Comment by nc-kano** [comment]
Currently we can only select images OR videos so also option for images AND videos would be nice.

**Comment by harima34** [comment]
Any news ?

**Comment by Brosten** [comment]
The camera and photo selection parts really needs to be better in MAUI. I just learned that another team close to me picked another framework in favor of .net MAUI with the lack of good photo support as the major reason. Too bad MediaGallery (https://github.com/dimonovdd/Xamarin.MediaGallery) seems abandoned. Last commit 10 months ago.

**Comment by shingming** [comment]
> The camera and photo selection parts really needs to be better in MAUI. I just learned that another team close to me picked another framework in favor of .net MAUI with the lack of good photo support as the major reason. Too bad MediaGallery (https://github.com/dimonovdd/Xamarin.MediaGallery) seems abandoned. Last commit 10 months ago. 


I agree the photo picker in MAUI is very needs to be enhanced. The MediaGallery you mentioned is not abandoned and is currently good to go in the MAUI project. (I use it in my MAUI in my current pj, too)
In current situation, I think MediaGallery is the good solution.

![image](https://github.com/dotnet/maui/assets/65477270/50a8b906-b2cd-4f5d-8e1b-e5abb4d9c22f)

**Comment by Simo4565** [comment]
I would also need this feature

**Comment by mnidhk** [comment]
is there a workaround or still no option to pick multiple images?

**Comment by mnidhk** [comment]
is there a workaround or still no option to pick multiple images?

**Comment by shingming** [comment]
@mnidhk 

Currently, I use this plugin in my project to implement the multiple selection, and this works fine for me: https://github.com/dimonovdd/Xamarin.MediaGallery

**Comment by Odaronil** [comment]
I need this too! Microsoft pay attention!

**Comment by johannes-steurer** [comment]
yes, definitly needed

**Comment by samanson123** [comment]
Would be a very useful built in option. I'm having to use Xamarin.MediaGallery as a workaround same as many others.

**Comment by hernandoz** [comment]
This is an essential functionality for most enterprise apps, this should be a priority for the MAUI Team .

**Comment by SteHiggins** [comment]
We really need this.  Is there an ETA?

**Comment by janseris** [comment]
> We really need this. Is there an ETA?

MAUI is dead

**Comment by mackayn** [comment]
Xamarin media gallery can be ported to .NET8 for now, just done this for a project to get multi-photos select unblocked.

https://github.com/dimonovdd/Xamarin.MediaGallery

**Comment by waldevelop** [comment]
Hi Mackayn
how di you convert it to .NET 8?  In Visual Studio I don't find this option and It's not possible to build it for Net 6

**Comment by samanson123** [comment]
> Hi Mackayn
> how di you convert it to .NET 8?  In Visual Studio I don't find this option and It's not possible to build it for Net 6

Downloaded the latest version of the Nuget package in your .NET 8 project.

**Comment by mackayn** [comment]
> > Hi Mackayn
> > how di you convert it to .NET 8?  In Visual Studio I don't find this option and It's not possible to build it for Net 6
> 
> Downloaded the latest version of the Nuget package in your .NET 8 project.

That didn't work for me anyway, it wouldn't compile. I manually changed the framework references to .NET8 in the proj file, there's a LOT of compilers redirections to support portable & .NET8, we made some small tweaks for deprecated iOS calls but it's not a package as such, just a solution project, also had to stub a few interface implementations to allow it to compile .NET8 so the app could be unit tested but if you don't have to do that, you don't need to worry about it.

**Comment by mackayn** [comment]
> > We really need this. Is there an ETA?
> 
> MAUI's development is ~dead Realistic perspective from a person who has been following it for 2 years (since 4 months before its release) and judging by what and how many issues have been or are being fixed and what embarrassing issues which we did not think were possible in a production UI framework our developers and me have been encountering and still are encountering). Fixes or new features which are released are negligible and take years even if they are important because there are few developers included and this has not changed much in these 2 years.

It has it's challenges but just finished porting a large enterprise SAAS app to MAUI successfully.

**Comment by mackayn** [comment]
> Hi Mackayn how di you convert it to .NET 8? In Visual Studio I don't find this option and It's not possible to build it for Net 6

I manually changed the csroj files and compiler redirections in code to .NET8. You need to get your hands dirty under the hood. This package could be updated pretty easily to support .NET8. There's a few other things that might need tweaking.

**Comment by endeakidz** [comment]
This should be the highest priority for MAUI iOS / Android development.

**Comment by MitchBomcanhao** [comment]
> This should be the highest priority for MAUI iOS / Android development.

This should be the highest priority for MAUI ~~iOS / Android~~ development.

**Comment by h0lg** [comment]
> It would also be nice to have a CapturePhotosAsync to take multiple photos without closing the camera

Agreed! It could be introduced into the UI flow as a third option of the dialog you get after taking a photo: `Retake` | `Use Photo` | **`Use and take another`**

**Comment by Alex-Dobrynin** [comment]
Hi guys, using this one lib for picking media files MPowerKit.MediaPlugin. Works like a charm. https://github.com/MPowerKit/MediaPlugin

**Comment by ichimukuren** [comment]
I support this issue. The ability to select multiple images in one action would be very helpful. Using FilePicker.PickMultipleAsync as a workaround adds unnecessary complexity, especially when only images are needed. Adding a method like PickMultiplePhotosAsync in the MediaPicker class would simplify this process significantly.

**Comment by h0lg** [comment]
You can currently **work around** this by re-triggering the taking of a single photo until the user cancels:

```c#
private async Task TakePhoto()
{
    var photo = await MediaPicker.Default.CapturePhotoAsync();

    if (photo != null)
    {
        await SavePhotoAsync(photo);
        await Task.Delay(1000); // seems to be required for MediaPicker to open again
        await TakePhoto(); // keep taking photos until user cancels
    }
}
```

**Comment by Alex-Dobrynin** [comment]
@h0lg if i would do this workaround, i probably could be fired few times

**Comment by h0lg** [comment]
> [@h0lg](https://github.com/h0lg) if i would do this workaround, i probably could be fired few times

@Alex-Dobrynin I think we can agree on that this is a bit of a cheeky work around :) Yet it achieves something close to the desired behaviour with the current MediaPicker API.
Obviously, it shows off capturing instead of picking - and you may want to consider making this mode an _alternative_ to picking a single photo instead of the default. Adapt as you like.

Send my regards to your boss. Pragmatism is not for everyone :)

**Comment by Simoo23** [comment]
I need this too!

**Comment by waldevelop** [comment]
I tried to install this plugin from NuGet  https://github.com/MPowerKit/MediaPlugin but it can't be installed on Windows. I can see it inside the IOS and Android packages after the installation but not inside Windows so I can't reference it inside the MauiProgram.cs file. Did someone experience the same problem ?

**Comment by Simoo23** [comment]
At the moment I'm using MediaGallery (https://github.com/dimonovdd/Xamarin.MediaGallery) but an official solution (with official support) will be better

**Comment by waldevelop** [comment]
thanks @Alex-Dobrynin I tried to follow the suggestions but the problem is still there. I'll try to see if a PC reboot could be the solution

**Comment by waldevelop** [comment]
I found the source of problem I think... the plugin can be installed in net8.0-windows 10.0.2261.45 at least, instead I was using net8.0-windows 10.0.19521
How can I update it ?

**Comment by waldevelop** [comment]
This plugin is working perfectly (https://github.com/dimonovdd/Xamarin.MediaGallery) is there also any option to translate the few words inside the Media Gallery opened ? I'd like to translate "Select photos", or "Done" )

**Comment by michalpr** [comment]
If you would like not to depend on third party package, you can use this simple workaround - I wrote short tutorial [here](https://medium.com/@michalpr/maui-mediapicker-multiple-selection-d0f678dcd5f7).

Solution, in short, uses iOS native class PHPickerViewController and connect its use with standard FilePicker class on Android and Windows platforms.

**Comment by Conbag93** [comment]
This is a shining example of how uninspiring the development of MAUI really is.

This should have been a feature on day one, yet here wae are nearly 3 years later with a bunch of hacky workarounds and zero ETA on an official solution. How could you design a "MediaPicker" class and have it _not_ occur to you that users might want to select more than one file.

**Comment by jfversluis** [comment]
Implemented through #29079
